### ISSUE TITLE
new-app command needs to support explicit contextDir setting

### ISSUE THREAD

**Issue by jpechane** [openshift/origin#1877]
Let's suppose I have a repository for STI build. If this repo contains more than one module (which is not uncommon in Maven builds) it is very useful to have an option to point to the contextDir that should be used for the build.

See https://github.com/jboss-developer/jboss-eap-quickstarts

For new users I'd like to have an option to issue command
osc new-apphttps://github.com/jboss-developer/jboss-eap-quickstarts --docker-image=ce-registry.usersys.redhat.com/jboss-eap6/eap-openshift:6.4 --contextDir=helloworld to run helloworld example on OpenShift without further changes

**Comment by kargakis** [comment]
My concern is that new-app can accept more than one source repositories:

```
[vagrant@openshiftdev sample-app]$ openshift cli new-app https://github.com/openshift/ruby-hello-world https://github.com/sspeiche/node-echo.git -l name=hello
imageStreams/ruby-hello-world
buildConfigs/ruby-hello-world
deploymentConfigs/ruby-hello-world
imageStreams/node-echo
buildConfigs/node-echo
services/ruby-hello-world
routes/ruby-hello-world
A build was created - you can run `osc start-build ruby-hello-world` to start it.
A build was created - you can run `osc start-build node-echo` to start it.
Service "ruby-hello-world" created at 172.30.210.50:8080,172.30.210.50:8080 to talk to pods over port(s) 8080,8080.
```

So if I specify --contextDir above once, in which repository should it be used?

**Comment by jpechane** [comment]
Could we provide sythetic url like https://github.com/openshift/ruby-hello-world#<contextDir> or https://github.com/openshift/ruby-hello-world?contextDir=<contextDir>?

Beware I do not know if git uses anchors and/or url paramters for its own purposes

**Comment by kargakis** [comment]
The short answer is yes but this is part of a bigger problem where we need a general scheme for components in new-app. See [this](https://github.com/openshift/origin/pull/1747#issuecomment-95326022).

**Comment by smarterclayton** [comment]
This was fixed by #2476
